### PR TITLE
Fix for REGISTRY-3379

### PR DESCRIPTION
**Pull request by Malintha**

### CODE REVIEW

File: apps/store/modules/page-decorators.js [hunk 1]
```diff
@@ -163,45 +163,83 @@ var pageDecorators = {};
             return permissions.hasAssetPermission(permissions.ASSET_BOOKMARK, type, tenantId, username);
         };
         var bookmarkPerms = {};
-        for (var index in types) {
-            typeDetails = ctx.rxtManager.getRxtTypeDetails(types[index]);
-            type = typeDetails.shortName;
+        
+        // check whether the given query is a mediaType search query. Due to REGISTRY-3379.
+        // case 1 : Search query provided with mediaType search
+        if(isMediaType(query,types)){
```
Comment (mushthaq33) [inline apps/store/modules/page-decorators.js]: Formatting issue

File: apps/store/modules/page-decorators.js [hunk 2]
```diff
@@ -163,45 +163,83 @@ var pageDecorators = {};
...
             }
-        }
         page.recentAssets = items;
         page.recentAssetsByType = assetsByType;
     };
+
+    /**
+     * Method to check whether a user has entered a mediaType search query.
+     */
+    var isMediaType = function(q,types){
```
Comment (mushthaq33) [inline apps/store/modules/page-decorators.js]: Formatting issue

File: apps/store/modules/page-decorators.js [hunk 3]
```diff
@@ -163,45 +163,83 @@ var pageDecorators = {};
...
+         var hasMediaType = q ? Boolean(q.mediaType) : false;
+         //if a query is not provided or if media type is not provided we will skip media scoping 
+         if(!hasMediaType) {
+            return hasMediaType;            
+         }
+         var mediaType = q.mediaType;
+        return types.filter(function(type){
```
Comment (daneshk) [inline apps/store/modules/page-decorators.js]: formatting issue

File: apps/store/modules/page-decorators.js (hunk 1 above)
Comment (Malintha) [inline apps/store/modules/page-decorators.js]: Fixed in https://github.com/wso2/carbon-store/pull/843

File: apps/store/modules/page-decorators.js (hunk 2 above)
Comment (Malintha) [inline apps/store/modules/page-decorators.js]: Fixed in https://github.com/wso2/carbon-store/pull/843

File: apps/store/modules/page-decorators.js (hunk 3 above)
Comment (Malintha) [inline apps/store/modules/page-decorators.js]: Fixed in https://github.com/wso2/carbon-store/pull/843

**Pull request merged**